core/src/main/scala/org/apache/spark/sql/delta/commands/VacuumCommand.scala

```diff
@@ -498,9 +498,45 @@ trait VacuumCommandImpl extends DeltaCommand {
       basePath: Path,
       relativizeIgnoreError: Boolean,
       isBloomFiltered: Boolean): Seq[String] = {
-    getActionRelativePath(action, fs, basePath, relativizeIgnoreError).map { relativePath =>
+    val paths = getActionRelativePath(action, fs, basePath, relativizeIgnoreError).map {
+      relativePath =>
       Seq(relativePath) ++ getAllSubdirs("/", relativePath, fs)
     }.getOrElse(Seq.empty)
+
+    val deletionVectorPath =
+      getDeletionVectorRelativePath(action, fs, basePath, relativizeIgnoreError).map(pathToString)
+
+    paths ++ deletionVectorPath.toSeq
+  }
+
+    /** Returns the relative path of the on-disk deletion vector. */
+  protected def getDeletionVectorRelativePath(
+      action: FileAction,
+      fs: FileSystem,
+      basePath: Path,
+      relativizeIgnoreError: Boolean
+  ): Option[Path] = {
+    val dv = action match {
+      case a: AddFile if a.deletionVector != null =>
+        Some(a.deletionVector)
+      case r: RemoveFile if r.deletionVector != null =>
+        Some(r.deletionVector)
```

Comment:
We should refactor `deletionVector` to `FileAction` to avoid having to write this kind of stuff. No need to be in this PR though.